src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -425,11 +439,43 @@ namespace winrt::Microsoft::Terminal::Control::implementation
                 return true;
             }
 
-            if (ch == Enter)
+            // If we have a connection info with which to try and re-create the
+            // connection, then let's try doing that on Enter
+            if (_ConnectionInfo != nullptr)
             {
-                _connection.Close();
-                _connection.Start();
-                return true;
+                if (ch == Enter)
+                {
+                    // Manually update the startingDirectory. If we had a CWD set
+                    // by the client, then let's try and use that as the CWD for
+                    // a restart, so it's more "seamless"
+                    _ConnectionInfo.Settings().Insert(L"startingDirectory", Windows::Foundation::PropertyValue::CreateString(WorkingDirectory()));
+                    // pass in the magic "inheritCursor" setting to the
+                    // connection's settings. This'll cause conpty to restart
+                    // the connection at the current place in the buffer.
+                    _ConnectionInfo.Settings().Insert(L"inheritCursor", Windows::Foundation::PropertyValue::CreateBoolean(true));
+                    auto c = TerminalConnection::ConnectionInformation::CreateConnection(_ConnectionInfo);

```

Comment:
okay, I see, but I still don't love it. "Sometimes the connection comes from the app and sometimes TermControl spontaneously ~~creates life and seeds it onto an empty planet~~ makes its own connection".